UpdateProfile/__tests__/handler.test.ts

```diff
@@ -847,4 +845,76 @@ describe("UpdateProfileHandler", () => {
 
     expect(mockSendMessage).toBeCalledTimes(0);
   });
+
+  it("GIVEN a profile with a valid last_app_version, the handler should write the field and return successfully", async () => {
+    const profileModelMock = {
+      findLastVersionByModelId: jest.fn(() => TE.of(some({ ...aRetrievedProfile, lastAppVersion: undefined }))),
+      update: jest.fn(_ => TE.of({ ...aRetrievedProfile, ..._ }))
+    };
+
+    const updateProfileHandler = UpdateProfileHandler(
+      profileModelMock as any,
+      mockQueueClient,
+      mockTracker
+    );
+
+    const result = await updateProfileHandler(contextMock as any, aFiscalCode, {
+      ...aProfile,
+      last_app_version: "0.0.1" as Semver
+    });
+
+    expect(result.kind).toBe("IResponseSuccessJson");
+    if (result.kind === "IResponseSuccessJson") {
+      expect(result.value.last_app_version).toBe("0.0.1");
+    }
+  })
+
+  it("GIVEN a profile with a valid last_app_version, the handler should overwrite the field and return successfully", async () => {
+    const profileModelMock = {
+      findLastVersionByModelId: jest.fn(() => TE.of(some({ ...aRetrievedProfile, lastAppVersion: "UNKNOWN" }))),
+      update: jest.fn(_ => TE.of({ ...aRetrievedProfile, ..._ }))
+    };
+
+    const updateProfileHandler = UpdateProfileHandler(
+      profileModelMock as any,
+      mockQueueClient,
+      mockTracker
+    );
+
+    const result = await updateProfileHandler(contextMock as any, aFiscalCode, {
+      ...aProfile,
+      last_app_version: "0.0.1" as Semver
+    });
+
+    expect(result.kind).toBe("IResponseSuccessJson");
+    if (result.kind === "IResponseSuccessJson") {
+      expect(result.value.last_app_version).toBe("0.0.1");
+    }
+  })
+
+  it("GIVEN a profile without last_app_version field, the update function will take that field as undefined", async () => {
+    const profileModelMock = {
+      findLastVersionByModelId: jest.fn(() => TE.of(some({ ...aRetrievedProfile, lastAppVersion: "0.0.1" }))),
```

Comment:
`lastAppVersion` override is not needed here, `aRetrievedProfile` already has it.
```suggestion
      findLastVersionByModelId: jest.fn(() => TE.of(some(aRetrievedProfile))),
```